vertx/src/main/java/io/undertow/vertx/VertxUndertowEngine.java

```diff
@@ -141,7 +151,13 @@ public void start(Future<Void> startFuture) throws Exception {
             server = vertx.createHttpServer(options);
 
             server.requestHandler(request -> {
-                VertxHttpExchange delegate = new VertxHttpExchange(request, allocator, blockingExecutor, null);
+                VertxHttpExchange delegate =
+                      new VertxHttpExchange(request, allocator, blockingExecutor, null);
+                if (blockingReadTimeout != null) {
+                    vertx.setTimer(blockingReadTimeout,
```

Comment:
This is not a read timeout, this is a 'max request time', in addition as the timer is not cancelled the connection will always be killed later.